app/Http/Controllers/Backend/PermissionsController.php

```diff
@@ -0,0 +1,297 @@
+<?php
+namespace App\Http\Controllers\Backend;
+
+use Illuminate\Http\Request;
+use App\Http\Requests;
+use App\Http\Controllers\Controller;
+use App\Models\Backend\Permission;
+
+class PermissionsController extends Controller
+{
+    /**
+     * Display a listing of the Permissions.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $data['data'] = Permission::all();
+        return view('backend.permissions.index')->with($data);
+    }
+
+    /**
+     * Show the form for creating a new Permissions.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $modules['modules'] = $this->getModule();
+        return view('backend.permissions.create')->with($modules);
+    }
+
+    /**
+     * Store a newly created Permissions in storage.
+     *
+     * @param \Illuminate\Http\Request $request request create
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        // check permission is exists?
+        if (!$this->checkPermissionExisted($request)) {
+            // create new permission
+            $newRole = new Permission;
```

Comment:
This code is too long and unnecessary. Just
```php
$arrRole = $this->setPermission($request);
$newRole = new Permission($arrRole);
$newRole->module = $request->module;
$newRole->save();
```
No need to assign property one by one.